app/Http/Controllers/Event/EventController.php

```diff
@@ -32,9 +32,11 @@ public function __construct()
         ]]);
 
         $this->middleware('auth_organization', ['only' => [
-            'create', 'store', 'edit', 'update', 'destroy',
+            'create', 'store', 'edit', 'update', 'organization_cancel'
         ]]);
-    }
+
+		$this->middleware('auth_admin', ['only' => ['admin_cancel', 'destroy']]);
```

Comment:
I told you we have a cancel function for organizations and destroy function for admins